sycl/source/detail/pi_opencl.cpp

```diff
@@ -159,6 +159,33 @@ pi_program OCL(piProgramCreate)(pi_context context, const void *il,
   return pi_cast<pi_program>(resProgram);
 }
 
+pi_sampler OCL(piSamplerCreate)(pi_context context,
+                                const cl_sampler_properties *sampler_properties,
+                                pi_result *error_code) {
+  cl_sampler returnSampler;
+  cl_bool normalizedCoords;
+  cl_addressing_mode addressingMode;
+  cl_filter_mode filterMode;
+
+  // Unpack sampler properties
+  for (std::size_t i = 0; sampler_properties && sampler_properties[i] != 0;
+       ++i) {
+    if (sampler_properties[i] == CL_SAMPLER_NORMALIZED_COORDS) {
+      normalizedCoords = sampler_properties[++i];
+    } else if (sampler_properties[i] == CL_SAMPLER_ADDRESSING_MODE) {
+      addressingMode = sampler_properties[++i];
+    } else if (sampler_properties[i] == CL_SAMPLER_FILTER_MODE) {
+      filterMode = sampler_properties[++i];
+    }
```

Comment:
Should I add another `else` branch that indicates that property name wasn't found?

Reply:
sure, that would be more robust. you can probably use PI_ASSERT